### PR TITLE
Add model control plane - release

### PR DESCRIPTION
**Pull request by safoinme**

### CODE REVIEW

File: template/steps/promotion/{% if metric_compare_promotion %}promote_get_metrics.py{% endif %} [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+from zenml.enums import ModelStages
+
+logger = get_logger(__name__)
+
+model_registry = Client().active_stack.model_registry
+
+@step
+def promote_get_metrics(
+) -> (
+    Tuple[Annotated[dict[str, Any], "latest_metrics"], Annotated[dict[str, Any], "current_metrics`"]]
```
Comment (htahir1) [inline template/steps/promotion/{% if metric_compare_promotion %}promote_get_metrics.py{% endif %}:17]: Should be `Dict`

File: template/steps/promotion/{% if metric_compare_promotion %}promote_metric_compare_promoter.py{% endif %} [hunk 2]
```diff
@@ -13,10 +14,9 @@ model_registry = Client().active_stack.model_registry
 
 @step
 def promote_metric_compare_promoter(
-    latest_metric: float,
-    current_metric: float,
-    latest_version: str,
-    current_version: str,
+    latest_metrics: dict[str, str],
```
Comment (htahir1) [inline template/steps/promotion/{% if metric_compare_promotion %}promote_metric_compare_promoter.py{% endif %}:17]: `Dict` not dict

File: template/steps/promotion/{% if metric_compare_promotion %}promote_metric_compare_promoter.py{% endif %} [hunk 3]
```diff
@@ -13,10 +14,9 @@ model_registry = Client().active_stack.model_registry
...
+    current_metrics: dict[str, str],
```
Comment (htahir1) [inline template/steps/promotion/{% if metric_compare_promotion %}promote_metric_compare_promoter.py{% endif %}:18]: `Dict` not `dict`

File: template/steps/promotion/{% if metric_compare_promotion %}promote_metric_compare_promoter.py{% endif %} [hunk 4]
```diff
@@ -37,50 +37,41 @@ def promote_metric_compare_promoter(
...
+        current_metrics: Previously promoted model metrics results.
     """
 
+
     ### ADD YOUR OWN CODE HERE - THIS IS JUST AN EXAMPLE ###
     pipeline_extra = get_step_context().pipeline_run.config.extra
     should_promote = True
 
-    if latest_version == current_version:
+    breakpoint()
```
Comment (htahir1) [inline template/steps/promotion/{% if metric_compare_promotion %}promote_metric_compare_promoter.py{% endif %}:49]: breakpoint? :-D

File: template/steps/promotion/{% if metric_compare_promotion %}promote_metric_compare_promoter.py{% endif %} [hunk 5]
```diff
@@ -37,50 +37,41 @@ def promote_metric_compare_promoter(
...
-        latest_metric: Recently trained model metric results.
-        current_metric: Previously promoted model metric results.
-        latest_version: Recently trained model version.
-        current_version: Previously promoted model version.
-
+        latest_metrics: Recently trained model metrics results.
+        current_metrics: Previously promoted model metrics results.
     """
 
+
```
Comment (htahir1) [inline template/steps/promotion/{% if metric_compare_promotion %}promote_metric_compare_promoter.py{% endif %}:44]: extra line

**Review by htahir1 — CHANGES_REQUESTED** [review]
Found some wrongly typed things like `Dict` not `dict` but im sure i missed some. Please go through all of these to make sure python 38 is is supported

File: template/pipelines/training.py [hunk 6]
```diff
@@ -12,7 +12,7 @@
     model_trainer,
     register_model,
 {%- if metric_compare_promotion %}
-    promote_get_metric,
+    promote_get_metrics,
```
Comment (htahir1) [inline template/pipelines/training.py:15]: why is this needed in the training pipeline?

File: template/config.yaml [hunk 7]
```diff
@@ -4,6 +4,8 @@ settings:
   docker:
 {%- if accelerator == 'gpu' %}
     parent_image: 'huggingface/transformers-pytorch-gpu'
+    build_options:
```
Comment (htahir1) [inline template/config.yaml:7]: is this needed?

**Review by htahir1 — CHANGES_REQUESTED** [review]

File: template/config.yaml (hunk 7 above)
Comment (htahir1) [reply template/config.yaml:7]: This doesnt work either it has to be 

```yaml
    build_options: 
      platform: 'linux/amd64'
```

Comment (safoinme) [reply template/config.yaml:7]: Unfortunately yes for m1 users, because it tries to pull an arm image which doesn't exist

Comment (htahir1) [reply template/config.yaml:7]: yes but still the actual command is wrong i think (see second comment)

**Review by htahir1 — CHANGES_REQUESTED** [review]
Great!

**Review by htahir1 — APPROVED** [review]

**Review by htahir1 — APPROVED** [review]
